components/MapViewScreen/index.tsx

```diff
@@ -51,20 +51,63 @@ export default function MapViewScreen({
     projectSize: { min: 0, max: 0 },
     location: [],
   });
+  const [filteredProjectsFromDropdowns, setFilteredProjectsFromDropdowns] =
+    useState<Project[]>(projects);
 
-  const handleFilterChange = (filter: FilterType) => {
-    console.log(filter);
+  const [filteredProjectsFromSearch, setFilteredProjectsFromSearch] =
+    useState<Project[]>(projects);
+
+  // clear filters
+  const clearFilters = () => {
+    setSelectedFilters({
+      status: [],
+      technology: [],
+      projectSize: { min: 0, max: 0 },
+      location: [],
+    });
+    setFilteredProjects(projects);
+  };
+
+  // show projects based on selected filters
+  const handleFilterButtonClick = () => {
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    const { status, technology, projectSize, location } = selectedFilters;
+    // add all filtering logic here
+    const technologyProjects = projects?.filter(project =>
+      technology.includes(project.renewable_energy_technology),
+    );
+    setFilteredProjectsFromDropdowns(technologyProjects);
   };
 
+  // search within all projects or filtered projects from dropdowns
   useEffect(() => {
-    let filtered: Project[] = [];
-    filtered =
-      projects?.filter(project =>
+    let projectsToSearch: Project[] = [];
+
+    if (filteredProjectsFromDropdowns.length > 0) {
+      projectsToSearch = filteredProjectsFromDropdowns;
+    } else {
+      projectsToSearch = projects;
+    }
+
+    const searchedProjects: Project[] =
+      projectsToSearch?.filter(project =>
```

Comment:
Can filters be applied after a search is made? In that case, would it make sense to search then filter?

I wonder if there's a way to treat "search" similar to a filter criteria. Each filter would then be applied sequentially in the order they're selected.

Reply:
The way that I've implemented it allows for filtering after a search is made! I think it does make sense to search then filter? For example, if someone wanted to checkout a project that has a common word "north", "south", etc. and then wanted to filter by dropdowns. This is probably a niche, rare use case but the way that the filtering is implemented allows for search then filter and filter then search to produce the same outputs in the project modal.

Do you mind explaining your idea of sequentially applying filters? I'm not quite sure I understand how the projects would be rendered differently.

Reply:
Ok it seems that the implementation already works as intended then. I was just curious about the ordering that filters are applied, because it could affect performance. 

For example, if you wish to search for "north" and filter by "hydroelectric," it might make sense to filter first then search, because depending on the search implementation, I imagine search would be an asymptotically slower operation (??), and thus better to perform on a smaller subset of projects.

The more I think about it, this is entirely dependent on search logic and how project data is stored, which I don't have much context on. Just something to consider!

Reply:
Hmm are you both suggesting that we prevent any searching until the user clicks on at least one filter? 

Regarding the time complexity of searching, I believe the way that we have set it up has filtering for search and dropdown filters to have the same complexity. We're iterating over each project and checking if it includes the search term or is included in the dropdown filters.

Reply:
I see that search isn't very complex in this case. Although I think the reason I bring this up is based off the main `useEffect`, where `projectsToSearch` selects all filtered projects, then the search term is applied to this list. Suppose I have filtered by `hydroelectric` then searched for `"north"`. I'd now have a list of projects satisfying these criteria. If I additionally applied the `operational` filter, ideally I'd take the list of already filtered projects, and apply the additional filter there. However, it seems that filters are applied first, separately from search, then the search term is applied after. Do you see what I mean? I'm not sure if that makes sense or whether it's actually needed, but something I noticed.

Reply:
Ohh ok I see what you are saying! I agree that we should do this for filters part of the `Status` filter. However, I don't think we can do this for a flow like `hydroelectric` -> search for `north` -> `land-based wind` (aka another `technology` filter). We would need to filter all projects based on the `technology` filters and those wouldn't exist in the modal prior to selecting `land-based wind`. Perhaps this is a task that Neha's recent PR can reflect then?

Reply:
For now, just squash and merge your sprint so that I can check everything works smoothly for this week's sprints. We can sort this out another time.